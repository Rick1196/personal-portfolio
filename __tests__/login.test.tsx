import React from "react";
import { render, screen } from "@utils/test-utils";
import renderer from "react-test-renderer";
import Login from "@components/login";

describe("Login component", () => {
  it("Login must render", () => {
    render(<Login />);
    expect(screen.getByText("Login with")).toBeInTheDocument();
  });

  it("Renders correctly", () => {
    // Updated test case with a Link to a different address
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
