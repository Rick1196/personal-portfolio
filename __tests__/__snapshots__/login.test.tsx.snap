// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Renders correctly 1`] = `
<div
  className="MuiContainer-root MuiContainer-maxWidthLg loginstyle__CenteredContainer-sc-t3qsyq-0 eLtxKO css-1oqqzyl-MuiContainer-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root loginstyle__CenteredBox-sc-t3qsyq-1 eCSXUg css-bhp9pd-MuiPaper-root-MuiCard-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Login with
    </h6>
    <button
      className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GoogleIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.545,10.239v3.821h5.445c-0.712,2.315-2.647,3.972-5.445,3.972c-3.332,0-6.033-2.701-6.033-6.032s2.701-6.032,6.033-6.032c1.498,0,2.866,0.549,3.921,1.453l2.814-2.814C17.503,2.988,15.139,2,12.545,2C7.021,2,2.543,6.477,2.543,12s4.478,10,10.002,10c8.396,0,10.249-7.85,9.426-11.748L12.545,10.239z"
          />
        </svg>
      </span>
      Google
    </button>
  </div>
</div>
`;
